cpu/nrf5x_common/clock.c

```diff
@@ -39,14 +43,34 @@ void clock_init_hf(void)
 #endif
 #endif
 
-#if CLOCK_HFCLK
-    /* start the HF clock */
-    NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
-    NRF_CLOCK->TASKS_HFCLKSTART = 1;
-    while (NRF_CLOCK->EVENTS_HFCLKSTARTED == 0) {}
+    /* allow to always enable the HFXO as non-default option */
+#if CLOCK_HFXO_ONBOOT
+    clock_hfxo_request();
 #endif
 }
 
+void clock_hfxo_request(void)
+{
+    unsigned state = irq_disable();
+    ++_hfxo_requests;
+    if (_hfxo_requests == 1) {
+        NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
+        NRF_CLOCK->TASKS_HFCLKSTART = 1;
+        while (NRF_CLOCK->EVENTS_HFCLKSTARTED == 0) {}
+    }
+    irq_restore(state);
+}
+
+void clock_hfxo_release(void)
+{
+    unsigned state = irq_disable();
+    --_hfxo_requests;
+    if (_hfxo_requests == 0) {
```

Comment:
```suggestion
    if (_hfxo_requests && --_hfxo_requests == 0) {
```

to avoid the possibility of underflow.

Reply:
I don't like this at all, as it hides inconsistent state in other modules/drivers and thus potentially leads to pretty ugly latent failures (see comment above)... 

However putting an assertion for `assert(_hfxo_requests);` in here would certainly not hurt!